src/components/Quiz/QuizWidget.tsx

```diff
@@ -137,94 +172,132 @@ const QuizWidget: React.FC<IProps> = ({ quizKey, maxQuestions }) => {
 
   const showConfetti = useMemo<boolean>(
     () => !!quizData && showResults && isPassingScore,
     [quizData, showResults, isPassingScore]
   )
 
-  // Handlers
-  const handleSelectAnswerChoice = (answerId: string): void => {
+  const handleSelectAnswerChoice = (answerId: string) => {
     const isCorrect =
       answerId === quizData?.questions[currentQuestionIndex].correctAnswerId
     setCurrentQuestionAnswerChoice({ answerId, isCorrect })
   }
 
-  const handleSelection = (answerId: string): void => {
+  const handleSelection = (answerId: string) => {
     setSelectedAnswer(answerId)
     handleSelectAnswerChoice(answerId)
   }
 
-  const handleShowAnswer = (questionId: string, answer: AnswerChoice): void => {
+  const handleShowAnswer = (questionId: string, answer: AnswerChoice) => {
     trackCustomEvent({
       eventCategory: "Quiz widget",
       eventAction: "Question answered",
       eventName: `QID: ${questionId}`,
       eventValue: answer.isCorrect ? "1" : "0",
     })
+
     setShowAnswer(true)
+
+    if (!isStandaloneQuiz) {
+      if (currentQuestionAnswerChoice?.isCorrect) {
+        statusHandler("success")
+      }
+
+      if (!currentQuestionAnswerChoice?.isCorrect) {
+        statusHandler("error")
+      }
+    }
   }
 
-  const handleRetryQuestion = (): void => {
+  const handleRetryQuestion = () => {
     trackCustomEvent({
       eventCategory: "Quiz widget",
       eventAction: "Other",
       eventName: "Retry question",
     })
+
     setCurrentQuestionAnswerChoice(null)
     setSelectedAnswer(null)
     setShowAnswer(false)
+
+    if (!isStandaloneQuiz) {
+      statusHandler("neutral")
+    }
   }
 
-  const handleContinue = (): void => {
+  const handleShare = () => {
+    if (!quizData || !window) return
+    trackCustomEvent({
+      eventCategory: "Quiz widget",
+      eventAction: "Other",
+      eventName: "Share results",
+    })
+
+    const url = `https://ethereum.org${window.location.pathname}%23quiz`
+    const hashtags = ["ethereumquiz", "ethereum", "quiz"]
+    const tweet = `${encodeURI(
+      `I just took the "${quizData.title}" quiz on ethereum.org and scored ${numberOfCorrectAnswers} out of ${quizData.questions.length}! Try it yourself at ${url}`
+    )}`
+
+    window.open(
+      `https://twitter.com/intent/tweet?text=${tweet}&hashtags=${hashtags}`
+    )
+  }
+
+  const handleContinue = () => {
     if (!currentQuestionAnswerChoice) return
+
     setUserQuizProgress((prev) => [...prev, currentQuestionAnswerChoice])
     setCurrentQuestionAnswerChoice(null)
     setShowAnswer(false)
-    if (showResults) {
+
+    // Reset quiz status (modifies bg color for mobile)
+    if (!isStandaloneQuiz) {
+      statusHandler("neutral")
+    }
+
+    if (finishedQuiz) {
       trackCustomEvent({
         eventCategory: "Quiz widget",
         eventAction: "Other",
         eventName: "Submit results",
-        eventValue: `${score}%`,
+        eventValue: `${quizScore}%`,
       })
     }
   }
 
```

Comment:
Should we update user stats when finishedQuiz is true?

If a user doesn't click 'Next quiz', their score doesn't update. So if they share on Twitter, x or click out of the modal, or try again, their score isn't saved. @konopkja any thoughts?

<img width="730" alt="Screenshot 2023-05-29 at 10 07 20" src="https://github.com/ethereum/ethereum-org-website/assets/62268199/82710b4d-49c9-40ab-bcfe-5147f5b253ac">

Reply:
> Should we update user stats when finishedQuiz is true?
> 
> If a user doesn't click 'Next quiz', their score doesn't update. So if they share on Twitter, x or click out of the modal, or try again, their score isn't saved. @konopkja any thoughts?
> 
> <img alt="Screenshot 2023-05-29 at 10 07 20" width="730" src="https://user-images.githubusercontent.com/62268199/241694171-82710b4d-49c9-40ab-bcfe-5147f5b253ac.png">

currently it's calculated on `Next Quiz` click only but the idea (still working on the fix) is to calculate when users click on `See results` (previous view). The thing is the logic of `showResults` from current implementation doesnt allow for this so I have to rewrite it (`finishedQuiz` was just a path)

Reply:
hm the score should be saved by the time user gets to this page ideally

Reply:
This is fixed now